core/builtin.py

```diff
@@ -485,18 +485,37 @@ def Cd(argv, mem):
 
 def Pushd(argv, dir_stack):
   dir_stack.append(os.getcwd())
+  num_args = len(argv)
+
+  if num_args <= 0:
+    log("pushd: no other directory")
+    return 1
+  elif num_args > 1:
+    log("pushd: too many arguments")
+
   dest_dir = argv[0]
-  os.chdir(dest_dir)  # TODO: error checking
+  if os.path.isdir(dest_dir):
+    os.chdir(dest_dir)
+  else:
+    log("pushd: " + dest_dir + ": No such file or directory")
```

Comment:
Please use

    util.error('pushd: %r: message', dest_dir)

The code is not super consistent, but `util.error()` is for fatal errors.

- Using the % style is preferred over +
- Be consistent with the rest of the file and use single quotes everywhere